link.go

```diff
@@ -71,41 +71,163 @@ type link struct {
 	msg                   Message             // current message being decoded
 }
 
-func newLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
+// newSendingLinkd creates a new sending link and attaches it to the session
+func newSendingLink(target string, s *Session, opts *SenderOptions) (*link, error) {
 	l := &link{
-		Key:                      linkKey{randString(40), encoding.Role(r != nil)},
+		Key:                      linkKey{randString(40), encoding.RoleSender},
 		Session:                  s,
-		receiver:                 r,
 		close:                    make(chan struct{}),
 		Detached:                 make(chan struct{}),
 		ReceiverReady:            make(chan struct{}, 1),
 		detachOnDispositionError: true,
+		Target:                   &frames.Target{Address: target},
+		Source:                   new(frames.Source),
+	}
+
+	if opts == nil {
+		return l, nil
 	}
 
-	// configure options
-	for _, o := range opts {
-		err := o(l)
-		if err != nil {
+	for _, v := range opts.Capabilities {
+		l.Source.Capabilities = append(l.Source.Capabilities, encoding.Symbol(v))
+	}
+	if opts.Durability > DurabilityUnsettledState {
+		return nil, fmt.Errorf("invalid Durability %d", opts.Durability)
+	}
+	l.Source.Durable = opts.Durability
+	if opts.DynamicAddress {
+		l.Target.Address = ""
+		l.dynamicAddr = opts.DynamicAddress
+	}
+	if opts.ExpiryPolicy != "" {
+		if err := encoding.ValidateExpiryPolicy(opts.ExpiryPolicy); err != nil {
 			return nil, err
 		}
+		l.Source.ExpiryPolicy = opts.ExpiryPolicy
+	}
+	l.Source.Timeout = opts.ExpiryTimeout
```

Comment:
I like the rename here, it makes it more obvious that this and the ExpiryPolicy are tied together.